servatrice/src/serversocketinterface.cpp

```diff
@@ -747,6 +749,45 @@ Response::ResponseCode ServerSocketInterface::cmdReplayDeleteMatch(const Command
 // MODERATOR FUNCTIONS.
 // May be called by admins and moderators. Permission is checked by the calling function.
 
+Response::ResponseCode ServerSocketInterface::cmdGetBanHistory(const Command_GetBanHistory &cmd, ResponseContainer &rc)
+{
+    if (!sqlInterface->checkSql())
+        return Response::RespInternalError;
+
+    QString userName = QString::fromStdString(cmd.user_name());
+
+    //lookup users ban history from database
+    QSqlQuery *query = sqlInterface->prepareQuery("SELECT A.id_admin, A.time_from, A.minutes, A.reason, A.visible_reason, B.name AS name_admin FROM {prefix}_bans A LEFT JOIN {prefix}_users B ON A.id_admin=B.id WHERE A.user_name = :user_name");
```

Comment:
The method being inside the socket interface?  I was discussing this with @ctrlaltca and the reason its done this way is the return response from the method would contain a fairly large amount of data.  If your returning one or two items no problem, but this is looking to grab 6 column items out of the table with the potential for a large repetitive number of those columns.  What type would you (or @ctrlaltca ) recommend I use to return such data?